core/Archive/ArchiveInvalidator.php

```diff
@@ -206,7 +213,11 @@ public function markArchivesAsInvalidated(array $idSites, array $dates, $period,
             $periodDates = $this->getPeriodDatesByYearMonthAndPeriodType($periods);
         }
 
+        // TODO: it's possible an archiving request could begin during invalidation, though probably unlikely that it would cause true disruption... just note in issue.
+
         $periodDates = $this->getUniqueDates($periodDates);
+
+        $idSites = $this->removeSitesWithInProgressArchiving($idSites);
```

Comment:
@diosmosis not sure I understand why we would filter the idSites here and not just invalidate all of them? Is this because we assume they were already invalidated / will be invalidated? Would there be any harm in still invalidating them or the issue that this would cause the remember flag to be removed?

Reply:
I wonder maybe we could still invalidate archives for the other idSites as well just not remove the remember flag?

Reply:
We remove them because we know they are currently being archived. We also keep the idSites in the remember flags so they will eventually be invalidated when they are not being archiving.

Reply:
Just wondering if there could be meanwhile some other previously finished archives for those idSites that need to be invalidated?

Reply:
There could be, but because of the way the function parameters are (idSites + list of dates + period type), it's a bit harder to control exactly what to invalidate. We'd have to loop over idSites & dates and check each individually, and invalidate each archive individually instead of with one query (which is what we do now I think).

Reply:
Not sure I understand... it would be safe to invalidate all the data related to that site/period and all parent periods as long as we keep the remember flag? Or would there be an issue?

Reply:
The issue is if we invalidate while archiving is ongoing, then it's possible for archives to re-initiate archiving for child periods re-doing archiving unnecessarily.

Eg, if the week period initiates archiving and finishes, then a day in the week is invalidated, and the month period starts archiving, it will re-archive the month and the day. Since we invalidate on new visits, this means for high traffic sites we'll always re-archive these subperiods as soon as an archive request goes through (since invalidation is not just done in core:archive but in Archive.php).

Reply:
Isn't that though what we still want it to behave? Whether it will be rearchived right away depends on `time_before_today_archive_considered_outdated` and `time_before_week_archive_considered_outdated` etc in that case or does it behave differently? I'm not sure how it behaves when data is invalidated... whether it still takes those values into consideration

Reply:
That's used by `getMinTimeProcessedForTemporaryArchive()` which is no longer used because we got rid of temporary archives... I guess we can re-add that logic?

Reply:
Incidentally I don't see `time_before_week_archive_considered_outdated` being used anywhere.

Reply:
@tsteur Ok I found where the period configs are used but they're not used anymore either because of the temporary archive change. This is likely the cause of the bug, and probably a less overarching change.

However, the lock does allow telling if a specific archive is being built (well not quite, but we can modify it if we increase the lock key column length). What do you think should happen? Make sure the time_before_ INI config is used now and keep the locks but don't do anything different in ArchiveInvalidator?

Reply:
@diosmosis it is very important we respect these configured values. We basically need to check if the last archive was done more than the configured time ago even if the archive is invalid (unless the archive is invalid and it is a new day I think then we don't respect these values but not sure? @mattab )

> and keep the locks but don't do anything different in ArchiveInvalidator?
I'm not quite fully understand this part but be good in general to maybe still invalidate an archive in ArchiveInvalidator? Not sure of consequences of not using Locks?

Reply:
> Not sure of consequences of not using Locks?

It's just an easy way to check if an archive is on-going no matter what machine/process it's on. It's not really necessary, but it might be useful in the future.

Reply:
I reckon be good to keep it 👍

Reply:
> It's just an easy way to check if an archive is on-going no matter what machine/process

@diosmosis it would be big value if it would help with this issue: https://github.com/matomo-org/matomo/issues/8444 -> maybe it would be fixed?

> unless the archive is invalid and it is a new day I think then we don't respect these values but not sure? @mattab

correct, these INI settings are only used when the date range includes Today.

Reply:
just wanting to make sure we're not going to work on #8444 as part of this issue :)